tests/test_strict_mode.py

```diff
@@ -0,0 +1,177 @@
+"""Tests for strict mode configuration.
+
+This module tests the codec_pipeline.strict configuration option that controls
+whether ZarrsCodecPipeline raises exceptions or falls back to BatchedCodecPipeline
+for unsupported operations.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pytest
+import zarr
+from zarr.storage import StorePath
+
+from zarrs.pipeline import (
+    # CollapsedDimensionError,
+    DiscontiguousArrayError,
+    UnsupportedDataTypeError,
+    # UnsupportedMetadataError,
+)
+
+if TYPE_CHECKING:
+    from zarr.abc.store import Store
+
+
+class TestStrictModeDisabled:
```

Comment:
I think these two sets of tests could be smashed together and parametrized by `codec_pipeline_strict` so something like:

```python
with pytest.raises(UnsupportedDataTypeError) if  codec_pipline_strict else null_context()
        arr[:] = data
if not codec_pipeline_strict:
        result = arr[:]
        assert np.array_equal(result, data)
```